src/main/java/soconnect/logic/parser/SearchCommandParser.java

```diff
@@ -54,6 +54,16 @@ public SearchCommand parse(String args) throws ParseException {
      */
     private boolean isValidSearchKeyword(ArgumentMultimap argMultimap) {
         List<String> keywords = argMultimap.getAllValues();
+        boolean valid = false;
+        for (String keyword : keywords) {
+            if (!keyword.trim().isEmpty()) {
+                valid = true;
+                break;
+            }
+        }
+        if (!valid) {
+            return false;
+        }
```

Comment:
Perhaps you could replace `valid = true` with `return false` and remove the `!` in `keyword.trim().isEmpty()`?